kubernetes/linux/main.sh

```diff
@@ -150,6 +150,17 @@ else
       echo "LA Onboarding:Workspace Id not mounted, skipping the telemetry check"
 fi
 
+# Set environment variable for if public cloud by checking the workspace domain.
+if [ -z $domain ]; then
+  ClOUD_ENVIRONMENT="unknown"
+elif [ $domain == "opinsights.azure.com" ]; then
```

Comment:
i think as we are starting to add more constants here, we can move the string constants to a file. Not in the scope of this PR. But we can discuss and make the changes later.